File: templates/languages.py
import requests
import os

langs_data={languages_json}


def UText(k):
    l = os.getenv("ULANG") if os.getenv("ULANG") else "default"
    if k in langs_data:
        if l in langs_data[k]:
            return langs_data[k][l] if langs_data[k][l] else k
        else:
            print(f"not supported lang: {l}")
            request_online(k,l)
            return k
    else:
        request_online(k,l)
        return k

def request_online(k,l):
    url = f"{api_addr}/{project_id}/key"  # Replace with the actual URL of your API endpoint

    params = {'k': k,'l':l}
    response = requests.get(url,  params=params)

    if response.status_code == 200:
        result = response.json()["result"]
        return result
    else:
        print(f"Failed to get data. Status code: {response.status_code}")

File: templates/languages.js
var langs_data = {languages_json}

function requestOnline(k, l) {
    var url = "{api_addr}/{project_id}/key";

    var params = {
        k: k,
        l: l
    };
    fetch(url + "?" + new URLSearchParams(params))
        .then(response => {
            if (response.ok) {
                return response.json();
            } else {
                console.log("Failed to get data. Status code: " + response.status);
                return null;
            }
        })
        .then(data => {
            if (data) {
                var result = data.result;
                return result;
            } else {
                return k;
            }
        })
        .catch(error => console.log(error));
}

function UText(k) {
    var l = process.env.ULANG ? process.env.ULANG : "fr";
    if (langs_data.hasOwnProperty(k)) {
        if (langs_data[k].hasOwnProperty(l)) {
            return langs_data[k][l] ? langs_data[k][l] : k;
        } else {
            console.log("not supported lang: " + l);
            requestOnline(k, l);
            return k;
        }
    } else {
        requestOnline(k, l);
        return k;
    }
}